diff --git a/node_modules/@stardust-collective/dag4-keyring/cjs/kcs.d.ts b/node_modules/@stardust-collective/dag4-keyring/cjs/kcs.d.ts
index 2d95377..297bf15 100644
--- a/node_modules/@stardust-collective/dag4-keyring/cjs/kcs.d.ts
+++ b/node_modules/@stardust-collective/dag4-keyring/cjs/kcs.d.ts
@@ -8,7 +8,9 @@ export declare enum KeyringWalletType {
     CrossChainWallet = "CCW",
     MultiAccountWallet = "MAW",
     SingleAccountWallet = "SAW",
-    MultiKeyWallet = "MKW"
+    MultiKeyWallet = "MKW",
+    LedgerAccountWallet = "LAW",
+    BitfiAccountWallet  = "BAW"
 }
 export declare enum KeyringAssetType {
     DAG = "DAG",
@@ -35,6 +37,8 @@ export declare type KeyringWalletAccountState = {
     address: string;
     network?: KeyringNetwork;
     tokens?: string[];
+    publicKey?: string;
+    deviceId?: string;
 };
 export declare type KeyringRingSerialized = {
     network: KeyringNetwork;
diff --git a/node_modules/@stardust-collective/dag4-keyring/cjs/kcs.js b/node_modules/@stardust-collective/dag4-keyring/cjs/kcs.js
index 3ef97cc..445580c 100644
--- a/node_modules/@stardust-collective/dag4-keyring/cjs/kcs.js
+++ b/node_modules/@stardust-collective/dag4-keyring/cjs/kcs.js
@@ -13,6 +13,8 @@ var KeyringWalletType;
     KeyringWalletType["MultiAccountWallet"] = "MAW";
     KeyringWalletType["SingleAccountWallet"] = "SAW";
     KeyringWalletType["MultiKeyWallet"] = "MKW"; //Single Chain, Multiple Key accounts, MKW
+    KeyringWalletType["LedgerAccountWallet"] = "LAW";
+    KeyringWalletType["BitfiAccountWallet"] = "BAW";
 })(KeyringWalletType = exports.KeyringWalletType || (exports.KeyringWalletType = {}));
 var KeyringAssetType;
 (function (KeyringAssetType) {
diff --git a/node_modules/@stardust-collective/dag4-keyring/cjs/keyring-manager.d.ts b/node_modules/@stardust-collective/dag4-keyring/cjs/keyring-manager.d.ts
index 5cbcc8a..b95eb00 100644
--- a/node_modules/@stardust-collective/dag4-keyring/cjs/keyring-manager.d.ts
+++ b/node_modules/@stardust-collective/dag4-keyring/cjs/keyring-manager.d.ts
@@ -12,7 +12,7 @@ export declare class KeyringManager extends SafeEventEmitter {
     private wallets;
     private memStore;
     private password;
-    constructor();
+    constructor({encryptor}?: {encryptor?: any});
     isUnlocked(): boolean;
     generateSeedPhrase(): string;
     private fullUpdate;
diff --git a/node_modules/@stardust-collective/dag4-keyring/cjs/keyring-manager.js b/node_modules/@stardust-collective/dag4-keyring/cjs/keyring-manager.js
index d29201d..bde3af6 100644
--- a/node_modules/@stardust-collective/dag4-keyring/cjs/keyring-manager.js
+++ b/node_modules/@stardust-collective/dag4-keyring/cjs/keyring-manager.js
@@ -31,11 +31,11 @@ const wallets_1 = require("./wallets");
 const dag4 = __importStar(require("@stardust-collective/dag4-core"));
 const bip39_helper_1 = require("./bip39-helper");
 class KeyringManager extends safe_event_emitter_1.default {
-    constructor() {
+    constructor({ encryptor }) {
         super();
         //Encrypted State
         this.storage = dag4.dagDi.getStateStorageDb();
-        this.encryptor = new encryptor_1.Encryptor();
+        this.encryptor = encryptor || new encryptor_1.Encryptor();
         this.memStore = new obs_store_1.ObservableStore({
             isUnlocked: false,
             wallets: [],
@@ -139,6 +139,9 @@ class KeyringManager extends safe_event_emitter_1.default {
         return wallet;
     }
     logout() {
+        // Reset ID counter that used to enumerate wallet IDs.
+        multi_chain_wallet_1.resetSid();
+        single_account_wallet_1.resetSid();
         this.password = null;
         this.memStore.updateState({ isUnlocked: false });
         this.emit('lock');
@@ -199,7 +202,7 @@ class KeyringManager extends safe_event_emitter_1.default {
         return this.findAccount(msgParams.from).signMessage(msgParams.from, msgParams.data, opts);
     }
     async unlockWallets(password) {
-        const encryptedVault = this.storage.get('vault');
+        const encryptedVault = await this.storage.get('vault');
         if (!encryptedVault) {
             //Support recovering wallets from migration
             this.password = password;
@@ -247,7 +250,7 @@ class KeyringManager extends safe_event_emitter_1.default {
         this.password = password;
         const sWallets = this.wallets.map(w => w.serialize());
         const encryptedString = await this.encryptor.encrypt(this.password, { wallets: sWallets });
-        this.storage.set('vault', encryptedString);
+        await this.storage.set('vault', encryptedString);
     }
     async _restoreWallet(wData) {
         let wallet;
diff --git a/node_modules/@stardust-collective/dag4-keyring/cjs/wallets/multi-chain-wallet.js b/node_modules/@stardust-collective/dag4-keyring/cjs/wallets/multi-chain-wallet.js
index 37762fa..8412d12 100644
--- a/node_modules/@stardust-collective/dag4-keyring/cjs/wallets/multi-chain-wallet.js
+++ b/node_modules/@stardust-collective/dag4-keyring/cjs/wallets/multi-chain-wallet.js
@@ -83,4 +83,5 @@ class MultiChainWallet {
     }
 }
 exports.MultiChainWallet = MultiChainWallet;
+exports.resetSid = () => SID = 0;
 //# sourceMappingURL=multi-chain-wallet.js.map
\ No newline at end of file
diff --git a/node_modules/@stardust-collective/dag4-keyring/cjs/wallets/single-account-wallet.js b/node_modules/@stardust-collective/dag4-keyring/cjs/wallets/single-account-wallet.js
index 3178711..371a4b4 100644
--- a/node_modules/@stardust-collective/dag4-keyring/cjs/wallets/single-account-wallet.js
+++ b/node_modules/@stardust-collective/dag4-keyring/cjs/wallets/single-account-wallet.js
@@ -84,4 +84,5 @@ class SingleAccountWallet {
     }
 }
 exports.SingleAccountWallet = SingleAccountWallet;
+exports.resetSid = () => SID = 0;
 //# sourceMappingURL=single-account-wallet.js.map
\ No newline at end of file
diff --git a/node_modules/@stardust-collective/dag4-keyring/esm/kcs.d.ts b/node_modules/@stardust-collective/dag4-keyring/esm/kcs.d.ts
index 2d95377..bed603f 100644
--- a/node_modules/@stardust-collective/dag4-keyring/esm/kcs.d.ts
+++ b/node_modules/@stardust-collective/dag4-keyring/esm/kcs.d.ts
@@ -8,7 +8,9 @@ export declare enum KeyringWalletType {
     CrossChainWallet = "CCW",
     MultiAccountWallet = "MAW",
     SingleAccountWallet = "SAW",
-    MultiKeyWallet = "MKW"
+    MultiKeyWallet = "MKW",
+    LedgerAccountWallet = "LAW",
+    BitfiAccountWallet = "BAW"
 }
 export declare enum KeyringAssetType {
     DAG = "DAG",
diff --git a/node_modules/@stardust-collective/dag4-keyring/esm/kcs.js b/node_modules/@stardust-collective/dag4-keyring/esm/kcs.js
index 30a59bb..ed493cc 100644
--- a/node_modules/@stardust-collective/dag4-keyring/esm/kcs.js
+++ b/node_modules/@stardust-collective/dag4-keyring/esm/kcs.js
@@ -10,6 +10,8 @@ export var KeyringWalletType;
     KeyringWalletType["MultiAccountWallet"] = "MAW";
     KeyringWalletType["SingleAccountWallet"] = "SAW";
     KeyringWalletType["MultiKeyWallet"] = "MKW"; //Single Chain, Multiple Key accounts, MKW
+    KeyringWalletType["LedgerAccountWallet"] = "LAW";
+    KeyringWalletType["BitfiAccountWallet"] = "BAW";
 })(KeyringWalletType || (KeyringWalletType = {}));
 export var KeyringAssetType;
 (function (KeyringAssetType) {
diff --git a/node_modules/@stardust-collective/dag4-keyring/esm/keyring-manager.d.ts b/node_modules/@stardust-collective/dag4-keyring/esm/keyring-manager.d.ts
index 5cbcc8a..b95eb00 100644
--- a/node_modules/@stardust-collective/dag4-keyring/esm/keyring-manager.d.ts
+++ b/node_modules/@stardust-collective/dag4-keyring/esm/keyring-manager.d.ts
@@ -12,7 +12,7 @@ export declare class KeyringManager extends SafeEventEmitter {
     private wallets;
     private memStore;
     private password;
-    constructor();
+    constructor({encryptor}?: {encryptor?: any});
     isUnlocked(): boolean;
     generateSeedPhrase(): string;
     private fullUpdate;
diff --git a/node_modules/@stardust-collective/dag4-keyring/esm/keyring-manager.js b/node_modules/@stardust-collective/dag4-keyring/esm/keyring-manager.js
index 50e81b8..252e84f 100644
--- a/node_modules/@stardust-collective/dag4-keyring/esm/keyring-manager.js
+++ b/node_modules/@stardust-collective/dag4-keyring/esm/keyring-manager.js
@@ -6,11 +6,11 @@ import { MultiChainWallet, SingleAccountWallet, MultiAccountWallet, MultiKeyWall
 import * as dag4 from '@stardust-collective/dag4-core';
 import { Bip39Helper } from './bip39-helper';
 export class KeyringManager extends SafeEventEmitter {
-    constructor() {
+    constructor({encryptor}) {
         super();
         //Encrypted State
         this.storage = dag4.dagDi.getStateStorageDb();
-        this.encryptor = new Encryptor();
+        this.encryptor = encryptor || new Encryptor();
         this.memStore = new ObservableStore({
             isUnlocked: false,
             wallets: [],
