diff --git a/node_modules/@stardust-collective/dag4-keystore/esm/key-store.js b/node_modules/@stardust-collective/dag4-keystore/esm/key-store.js
index 0888c80..4e71f2f 100644
--- a/node_modules/@stardust-collective/dag4-keystore/esm/key-store.js
+++ b/node_modules/@stardust-collective/dag4-keystore/esm/key-store.js
@@ -117,6 +117,11 @@ export class KeyStore {
         const ecSigHex = Buffer.from(ecSig.toDER()).toString('hex');
         return ecSigHex;
     }
+    personalSign (privateKey, msg) {
+        const encodedMessage = Buffer.from(msg, 'utf-8').toString('base64')
+        const prefixedMessage = `\u0019Constellation Signed Message:\n${encodedMessage.length.toString()}\n${msg}`;
+        return this.sign(privateKey, prefixedMessage);
+    }
     verify(publicKey, msg, signature) {
         const sha512Hash = this.sha512(msg);
         const validSig = curve.verify(sha512Hash, signature, Buffer.from(publicKey, 'hex'));
